AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/IncomingAsyncSubscriptionEventPublisher.swift

```diff
@@ -23,47 +23,51 @@ final class IncomingAsyncSubscriptionEventPublisher: Cancellable {
     typealias Event = AsyncEvent<SubscriptionEvent<GraphQLResponse<Payload>>, Void, APIError>
 
     private var onCreateOperation: GraphQLSubscriptionOperation<Payload>?
-    private let onCreateListener: GraphQLSubscriptionOperation<Payload>.EventListener
+    private var onCreateListener: GraphQLSubscriptionOperation<Payload>.EventListener?
```

Comment:
Here and with other listeners: I'm not sure why this needs to be an optional var? I don't see that you unassign it anywhere?

Reply:
It is because we are using `self.` inside the closure.  Setting to var allows us use `self` prior to all the `let`'s being assigned.